orderbook/src/account_balances.rs

```diff
@@ -0,0 +1,195 @@
+use anyhow::{Context, Result};
+use ethcontract::{batch::CallBatch, errors::MethodError, Http, Web3};
+use futures::future::{join, join_all, ok, FutureExt as _};
+use std::collections::HashMap;
+use std::sync::Mutex;
+
+use contracts::IERC20;
+use primitive_types::{H160, U256};
+
+#[async_trait::async_trait]
+pub trait BalanceFetching: Send + Sync {
```

Comment:
Why a trait? Are we expecting to mock it or have alternative implementations? Don't have a problem with that just feel that if the reason is maybe having an alternative implementation in the future then it would be better to only create the trait then.